Jm config changes

jsphmrrw opened the pull request:
@marcotranchino @urbanautomaton I'd be interested to hear what you think about these changes. I'm specifically interested in:

1. Whether you think we need to keep any of the config stuff I've deleted. I've assumed a lot of the config was copied from the quick start guide and that we might not actually need a lot of it.

2. If there was a reason we didn't initially include the `var-simplesamlphp/metadata/saml20-idp-hosted.php` that was recommended by the quick start guide. I've put it in here, mostly because I wanted to set the `NameIDFormat` option. I'm going to investigate whether I can remove the other stuff in there.

Commit:
Simplify AssertionConsumerService config

[1] says that the value of the `AssertionConsumerService` option can be
"specified in one of several endpoint formats". The endpoint formats are
listed by [2], where one of the options is a single string. A single
string can be used when "there is only a single endpoint that uses the
default binding", which I think is true in our case.

Using a single string to configure the `AssertionConsumerService` option
is a lot simpler and doesn't seem to break anything.

[1] https://simplesamlphp.org/docs/stable/simplesamlphp-reference-sp-remote
[2] https://simplesamlphp.org/docs/stable/simplesamlphp-metadata-endpoints

Commit:
Populate nameid with mail attribute

We currently set the `NameIdFormat` option to the emailAddress URN. This
doesn't actually populate the `nameid` of a response with an email,
though: It only describes the format of the value.

To actually populate `nameid` with an email address, we need configure
the `simplesaml.nameidattribute` option to pull the value from the
relevant `mail` attribute.

Commit:
Force auth on each request

This configures SimpleSAML so that credentials must be entered for each
request. Previously, a user would only have to provide their credentials
the first time they were authorized.

This congiguration was recommended by Stephen Feest, who has found it
useful in the past.

Commit:
Remove unneeded service provider metadata

Most of these options were generated by the metadata converter in the
simpleSAMLphp web frontend. I don't think they are required, so I'm
removing them.

urbanautomaton commented:
Hey @jsphmrrw!

1. The deleted configuration was generated by taking the XML metadata from our app and plugging it into the metadata converter in the simpleSAMLphp web frontend. I don't have a strong opinion on whether it's necessary to always do this. If what you've got works, great.

2. I think we didn't include the `-hosted.php` config because there's already one in the default install and I don't think we needed to customise it (or at least, hadn't yet realised we needed to). So, laziness basically. Having our own seems like a good idea.

:shipit:

Commit:
Add IdP metadata

This is the metadata that is recommended in the SimpleSAML quick start
guide [1].

We would like to configure the `NameIDFormat` option in a future commit;
however, it is not possible to configure this option without also
configuring the values included here.

[1] https://simplesamlphp.org/docs/stable/simplesamlphp-idp
